### PR TITLE
Add instrumentation for net/http and net/httptrace

### PR DESCRIPTION
**Pull request by Aneurysm9**
This PR moves the `othttp` and `httptrace` instrumentations from the main repo, renaming them to conform to this repo's naming conventions.  It also moves the HTTP example from the main repo and uses it to illustrate each instrumentation separately.

Part of #158 and https://github.com/open-telemetry/opentelemetry-go/issues/974

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ Thumbs.db
 *.so
 coverage.*
 
+instrumentation/github.com/gocql/gocql/example/example
```
Comment (lizthegrey) [inline .gitignore:11]: Inadvertent change?

File: internal/metric/mock.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
+// Copyright The OpenTelemetry Authors
```
Comment (lizthegrey) [inline internal/metric/mock.go:1]: intentional to copy these over rather than leaving them in otel package?

File: .gitignore (hunk 1 above)
Comment (Aneurysm9) [reply .gitignore:11]: `make precommit` summons that file into existence and I'm pretty sure it shouldn't be committed.

File: internal/metric/mock.go (hunk 2 above)
Comment (Aneurysm9) [reply internal/metric/mock.go:1]: They're internal to the otel package and thus can't be used here.  https://github.com/open-telemetry/opentelemetry-go/issues/872 could perhaps provide an exported solution, but one doesn't exist yet.

File: .gitignore (hunk 1 above)
Comment (lizthegrey) [reply .gitignore:11]: odd. https://github.com/open-telemetry/opentelemetry-go-contrib/pull/137 doesn't seem to imply automatically building the example in that dir, wonder why it's doing that. it should write to $GOPATH/bin instead...

Comment (MrAlias) [reply .gitignore:11]: Yeah, I noticed this as well. Thanks for catching it here.

**Comment by MrAlias** [comment]
@lizthegrey did your questions get answered or did we need to follow up on them?  Wondering if I can merge this.

File: instrumentation/net/http/common.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const (
+	ReadBytesKey  = kv.Key("http.read_bytes")  // if anything was read from the request body, the total number of bytes read
+	ReadErrorKey  = kv.Key("http.read_error")  // If an error occurred while reading a request, the string of the error (io.EOF is not recorded)
+	WroteBytesKey = kv.Key("http.wrote_bytes") // if anything was written to the response writer, the total number of bytes written
+	WriteErrorKey = kv.Key("http.write_error") // if an error occurred while writing a reply, the string of the error (io.EOF is not recorded)
+)
+
+// Server HTTP metrics
+const (
+	RequestCount          = "http.server.request_count"           // Incoming request count total
```
Comment (jmacd) [inline instrumentation/net/http/common.go:33]: (Note: there is a _belief_ that we can configure the `http.server.duration` metric, which has a count, to generate this `http.server.request_count` metric. I realize that we should not get hung up on views, so I think this looks good as a set of metrics.

Comment (MrAlias) [reply instrumentation/net/http/common.go:33]: FWIW, I noticed that this `const` isn't actually used in the code. It was one of the things I was alluding to needing clean up after the move.

**Pull request merged**